### ISSUE TITLE
PixelShuffle backward lost gradient. 

### ISSUE THREAD

**Issue by ailzhang** [pytorch/xla#922]
This might be related to the as_strided view, but saving it here so that I check back after the fix is merged. 
```
import torch
import torch_xla
import torch.nn as nn
import pdb

device = 'xla:0'
input_cpu = torch.rand(1, 9, 4, 4, requires_grad=True)
input = input_cpu.to(device).detach()
input.requires_grad = True
m_cpu = nn.PixelShuffle(3)
m = m_cpu.to(device)
output_cpu = m_cpu(input_cpu)
output = m(input)
output_cpu.sum().backward()
output.sum().backward()
print(input_cpu.grad)
print(input.grad)
```

**Comment by ailzhang** [comment]
Similarly
```
import torch
import torch_xla
import torch.nn as nn
import pdb

device = 'xla:0'

input_cpu = torch.rand(1, 2, 2, requires_grad=True)
input = input_cpu.to(device).detach()
input.requires_grad = True
m_cpu = nn.RReLU()
m = m_cpu.to(device)
output_cpu = m_cpu(input_cpu)
output = m(input)
output_cpu.sum().backward()
output.sum().backward()
print(input_cpu.grad)
print(input.grad)
```

**Comment by ailzhang** [comment]
Hmmm the first repro seem to have been fixed but the second is not. I'll take a look when I have time
